.github/workflows/main.yml

```
@@ -159,6 +159,20 @@ jobs:
         rm -rf /tmp/.buildx-cache
         mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
+  miri:
+    name: Miri
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: true
+    - name: Install Rust (rustup)
+      run: rustup update nightly --no-self-update && rustup default nightly
+      shell: bash
+    - run: rustup component add miri
+    - run: cargo miri setup
```

Does miri benefit much from caching?

```suggestion
    - run: cargo miri setup
    - uses: Swatinem/rust-cache@v2
```

Also could you add Miri to the `success` job at the bottom, so branch protection blocks on it?

---

What exactly gets cached there?

---

It saves the Cargo cache and the target directory. Probably doesn't hurt

---

Caching the target dir seems pointless since we're using the latest nightly so it will be invalidated every night, but sure :shrug: